improver_tests/developer_tools/test_MOMetadataInterpreter.py

```diff
@@ -0,0 +1,405 @@
+    appropriate attribute"""
+    ensemble_cube.attributes["title"].replace("MOGREPS-UK", "IMPROVER")
+    interpreter.run(ensemble_cube)
+    assert interpreter.model == "MOGREPS-UK"
+
+
+def test_error_forbidden_cell_method(blended_probability_below_cube, interpreter):
+    """Test error raised when a forbidden cell method is present"""
+    blended_probability_below_cube.add_cell_method(
+        CellMethod(method="mean", coords="forecast_reference_time")
+    )
+    with pytest.raises(ValueError, match="Non-standard cell method"):
+        interpreter.run(blended_probability_below_cube)
+
+
+def test_error_missing_time_bounds(blended_probability_below_cube, interpreter):
+    """Test error raised if a minimum in time cube has no time bounds"""
+    blended_probability_below_cube.coord("time").bounds = None
+    with pytest.raises(ValueError, match="has no time bounds"):
+        interpreter.run(blended_probability_below_cube)
+
+
+def test_error_incorrect_time_bounds(blended_probability_below_cube, interpreter):
+    """Test error raised if time points are not equal to upper bounds"""
+    blended_probability_below_cube.coord("time").bounds = np.array(
+        blended_probability_below_cube.coord("time").bounds + 3600, dtype=np.int64
+    )
+    with pytest.raises(ValueError, match="points should be equal to upper bounds"):
+        interpreter.run(blended_probability_below_cube)
+
+
+def test_error_missing_spot_coords(blended_spot_median_cube, interpreter):
+    """Test error raised if a spot cube doesn't have all the expected metadata"""
+    blended_spot_median_cube.remove_coord("altitude")
+    with pytest.raises(ValueError, match="Missing one or more coordinates"):
+        interpreter.run(blended_spot_median_cube)
+
+
+def test_error_missing_coords(probability_above_cube, interpreter):
+    """Test error raised if an unblended cube doesn't have the expected time
+    coordinates"""
+    probability_above_cube.remove_coord("forecast_period")
+    with pytest.raises(ValueError, match="Missing one or more coordinates"):
+        interpreter.run(probability_above_cube)
+
+
+def test_error_missing_blended_coords(blended_probability_below_cube, interpreter):
+    """Test error raised if a blended cube doesn't have the expected time
+    coordinates"""
+    blended_probability_below_cube.remove_coord("blend_time")
+    with pytest.raises(ValueError, match="Missing one or more coordinates"):
+        interpreter.run(blended_probability_below_cube)
+
+
+def test_error_time_coordinate_units(probability_above_cube, interpreter):
+    """Test error raised if time coordinate units do not match the IMPROVER standard"""
+    probability_above_cube.coord("forecast_period").convert_units("hours")
+    with pytest.raises(ValueError, match="does not have required units"):
+        interpreter.run(probability_above_cube)
+
+
+def test_weather_code_success(wxcode_cube, interpreter):
```

Comment:
Could this test and the wind-direction-success test be moved up to the successful tests section. I think it would make following these tests easier.